### PR TITLE
Added `crop_segmentation_mask` op

### PR DESCRIPTION
**Pull request by vfdev-5**
Related to #5514

## Description:
- Added functional `crop_segmentation_mask` op
- Added tests

### Results on synthetic images/bboxes/segm mask:

<details>
<summary>
Code
</summary>

```python
import numpy as np

import torch
import torchvision
from torchvision.prototype import features
from torchvision.prototype.transforms.functional import crop_image_tensor, crop_bounding_box, crop_segmentation_mask

size = (64, 76)
# xyxy format
in_boxes = [
    [10, 15, 25, 35],
    [50, 5, 70, 22],
    [45, 46, 56, 62],
]
labels = [1, 2, 3]

im1 = 255 * np.ones(size + (3, ), dtype=np.uint8)
mask = np.zeros(size, dtype=np.int64)
for in_box, label in zip(in_boxes, labels):
    im1[in_box[1]:in_box[3], in_box[0]:in_box[2], :] = (127, 127, 127)
    mask[in_box[1]:in_box[3], in_box[0]:in_box[2]] = label
    
in_im = torch.tensor(im1).permute(2, 0, 1).view(1, 3, *size)

in_boxes = features.BoundingBox(
    in_boxes, format=features.BoundingBoxFormat.XYXY, image_size=size
)
in_mask = features.SegmentationMask(torch.tensor(mask)).view(1, *size)
    
top = -10
left = 5
height = size[0] - 15
width = size[1] + 20


out_boxes = crop_bounding_box(
    in_boxes, 
    in_boxes.format,
    top,
    left,
)

out_mask = crop_segmentation_mask(
    in_mask, 
    top,
    left,
    height,
    width,
)

out_im = crop_image_tensor(in_im, top, left, height, width)
```

</details>

image

### CODE REVIEW

**Review by datumbox — APPROVED** [review]
LGTM.

@pmeier thoughts on the test side?

**Comment by vfdev-5** [comment]
> @pmeier thoughts on the test side?

Let's see test side later once Philip coded his functional opinfo-like proposal.
